app/scripts/migrations/068.js

```diff
@@ -0,0 +1,60 @@
+import { cloneDeep } from 'lodash';
+
+const version = 68;
+
+function addUrlProtocolPrefix(urlString) {
+  if (!urlString.match(/(^http:\/\/)|(^https:\/\/)/u)) {
+    return `https://${urlString}`;
+  }
+  return urlString;
+}
+
+export default {
+  version,
+
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData);
+    versionedData.meta.version = version;
+    const state = versionedData.data;
+    const newState = transformState(state);
+    versionedData.data = newState;
+    return versionedData;
+  },
+};
+
+function transformState(state) {
+  const PreferencesController = state?.PreferencesController || {};
+  const preferences = PreferencesController.preferences || {};
+  const oldIpfsGateWay = preferences.ipfsGateway;
+
+  let newState;
+
+  if (oldIpfsGateWay && oldIpfsGateWay !== 'dweb.link') {
+    const newIpfsGateway = new URL(
+      `${addUrlProtocolPrefix(oldIpfsGateWay)}/ipfs/`,
```

Comment:
do we want to add `/ipfs/` to every gateway?

Reply:
Yes I believe we do for retrieving assets atleast. Might have to do a bit of research on this though.

Reply:
Here is a list of different gateways: https://ipfs.github.io/public-gateway-checker/

Maybe the addition of `/ipfs/` should be done in the functions/classes responsible for retrieving assets? Or maybe we do want to store a separate property after all, `nftMetadataGateway` or something like that

Reply:
this concern should be addressed here: https://github.com/MetaMask/metamask-extension/pull/12873/commits/c68a608351921e2e456bb7e345ae338f71d7bf4e